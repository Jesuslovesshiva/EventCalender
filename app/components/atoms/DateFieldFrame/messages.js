import {defineMessages} from 'react-intl'

const scope = 'atoms.date-field-frame'

export default defineMessages({
  toggle: {
    id: `${scope}.toggle`,
    defaultMessage: 'Kalendar ein-/ausschalten',
    description: 'aria label toggle calendar button'
  }
})
