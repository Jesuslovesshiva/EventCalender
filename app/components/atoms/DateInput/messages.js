import {defineMessages} from 'react-intl'

const scope = 'atoms.date-picker'

export default defineMessages({
  ddmmyyyy: {
    id: `${scope}.date`,
    defaultMessage: 'TT.MM.JJJJ',
    description: 'placeholder for format dd.MM.yyyy'
  }
})
