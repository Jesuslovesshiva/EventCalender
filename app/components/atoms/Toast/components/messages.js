import {defineMessages} from 'react-intl'

const scope = 'atoms.toast'

export default defineMessages({
  success: {
    id: `${scope}.success`,
    defaultMessage: 'Info'
  },
  error: {
    id: `${scope}.error`,
    defaultMessage: 'Error or Warning'
  },
  alert: {
    id: `${scope}.alert`,
    defaultMessage: 'alert'
  },
  action: {
    id: `${scope}.action`,
    defaultMessage: 'Action'
  }
})
