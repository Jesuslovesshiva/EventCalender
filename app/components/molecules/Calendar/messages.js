import {defineMessages} from 'react-intl'

const scope = 'molecules.calendar'

export default defineMessages({
  selectMonth: {
    id: `${scope}.selectMonth`,
    defaultMessage: 'Monat auswählen'
  },
  selectYear: {
    id: `${scope}.selectYear`,
    defaultMessage: 'Jahr auswählen'
  },
  nextMonth: {
    id: `${scope}.nextMonth`,
    defaultMessage: 'nächstes Monat',
    description: 'aria label nextMonth button'
  },
  prevMonth: {
    id: `${scope}.prevMonth`,
    defaultMessage: 'vorheriges Monat',
    description: 'aria label prevMonth button'
  }
})
