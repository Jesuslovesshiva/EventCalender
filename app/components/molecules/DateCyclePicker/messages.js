import {defineMessages} from 'react-intl'

const scope = 'molecules.date-cycle-picker'

export default defineMessages({
  from: {
    id: `${scope}.from`,
    defaultMessage: 'von',
    description: 'aria label from input'
  },
  to: {
    id: `${scope}.to`,
    defaultMessage: 'bis',
    description: 'aria label to input'
  },
  next: {
    id: `${scope}.next`,
    defaultMessage: 'n√§chste Woche',
    description: 'aria label next button'
  },
  prev: {
    id: `${scope}.prev`,
    defaultMessage: 'vorherige Woche',
    description: 'aria label prev button'
  }
})
