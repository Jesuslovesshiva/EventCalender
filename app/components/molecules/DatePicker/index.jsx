import FocusTrap from 'focus-trap-react'
import noop from 'lodash/noop'
import PropTypes from 'prop-types'
import {forwardRef, useCallback} from 'react'

import Popup from '@/components/atoms/Popup'
import useIsOpen from '@/hooks/use-is-open'
import usePopup from '@/hooks/use-popup'
import useSyncState from '@/hooks/use-sync-state'

import Field from './components/Field'
import Picker from './components/Picker'

const DatePicker = forwardRef((props, ref) => {
  const [isOpen, {close, open, toggle}] = useIsOpen()
  const [value, setValue] = useSyncState(props.value)
  const {popper, referenceRef, popperRef} = usePopup({
    placement: 'bottom-start'
  })

  const onChange = useCallback(
    newDate => {
      props.onChange(newDate)
      setValue(newDate)
    },
    [props.onChange, setValue]
  )

  return (
    <FocusTrap
      active={isOpen}
      focusTrapOptions={{
        clickOutsideDeactivates: true,
        initialFocus: false,
        onDeactivate: close,
        returnFocusOnDeactivate: false
      }}
    >
      <div>
        <Field
          ref={referenceRef}
          close={close}
          inputRef={ref}
          isOpen={isOpen}
          onChange={onChange}
          open={open}
          toggle={toggle}
          value={value}
        />

        <Popup
          // eslint-disable-next-line react/jsx-props-no-spreading
          {...popper.attributes.popper}
          ref={popperRef}
          style={popper.styles.popper}
          isOpen={isOpen}
        >
          <Picker
            close={close}
            getEventsForDate={props.getEventsForDate}
            onChange={onChange}
            value={value}
          />
        </Popup>
      </div>
    </FocusTrap>
  )
})

DatePicker.displayName = 'DatePicker'

DatePicker.propTypes = {
  getEventsForDate: PropTypes.func,
  onChange: PropTypes.func,
  value: PropTypes.instanceOf(Date)
}

DatePicker.defaultProps = {
  getEventsForDate: noop,
  onChange: noop,
  value: null
}

export default DatePicker
