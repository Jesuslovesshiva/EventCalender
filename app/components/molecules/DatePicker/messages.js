import {defineMessages} from 'react-intl'

const scope = 'molecules.date-picker'

export default defineMessages({
  date: {
    id: `${scope}.date`,
    defaultMessage: 'Datum',
    description: 'aria label date input'
  }
})
