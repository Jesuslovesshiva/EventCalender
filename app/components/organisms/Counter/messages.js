import {defineMessages} from 'react-intl'

const scope = 'organisms.counter'

export default defineMessages({
  count: {
    id: `${scope}.count`,
    defaultMessage: 'Zähler: {count}',
    description: 'count output'
  },
  reset: {
    id: `${scope}.reset`,
    defaultMessage: 'Zurücksetzen',
    description: 'reset cta'
  }
})
