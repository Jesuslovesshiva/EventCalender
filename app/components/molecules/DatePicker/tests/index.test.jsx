import {fireEvent, waitFor} from '@testing-library/react'

import dateFieldFrameMessages from '@/components/atoms/DateFieldFrame/messages'
import configureTestStore from '@/utils/test-utils/configure-test-store'
import render from '@/utils/test-utils/custom-render'

import DatePicker from '../index'
import messages from '../messages'

const {clearDispatch, options} = configureTestStore()
const renderComponent = props => render(<DatePicker {...props} />, options)

describe('DatePicker', () => {
  beforeAll(() => {
    jest.useFakeTimers().setSystemTime(new Date(2022, 2, 28).getTime())
  })

  beforeEach(clearDispatch)

  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should close picker after date select', async () => {
    const {getByText, getByLabelText, queryByRole} = renderComponent({
      value: new Date('2022-03-28')
    })
    const toggle = getByLabelText(dateFieldFrameMessages.toggle.defaultMessage)
    fireEvent.click(toggle)
    expect(queryByRole('dialog')).toBeInTheDocument()
    fireEvent.click(getByText('10').parentElement)
    await waitFor(() => expect(queryByRole('dialog')).not.toBeInTheDocument())
  })

  describe('events', () => {
    it('should call onChange prop on input change', () => {
      const onChange = jest.fn()
      const {getByLabelText} = renderComponent({onChange})
      const input = getByLabelText(messages.date.defaultMessage)
      fireEvent.change(input, {target: {value: '08.03.2022'}})
      fireEvent.blur(input)
      expect(onChange).toHaveBeenCalledWith(new Date('2022-03-08 00:00:00'))
    })

    it('should call onChange prop on calendar day click', () => {
      const onChange = jest.fn()
      const {getByLabelText, getByText} = renderComponent({onChange})
      const toggle = getByLabelText(
        dateFieldFrameMessages.toggle.defaultMessage
      )
      fireEvent.click(toggle)
      fireEvent.click(getByText('10').parentElement)
      expect(onChange).toHaveBeenCalledWith(new Date('2022-03-10 00:00:00'))
    })
  })
})
