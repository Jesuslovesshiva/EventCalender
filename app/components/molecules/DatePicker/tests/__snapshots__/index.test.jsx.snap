// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should match the snapshot 1`] = `
<div>
  <div
    class="border border-gray-medium divide-x divide-gray-medium rounded inline-flex transition cursor-pointer hover:border-white hover:divide-white hover:bg-silver hover:shadow-md specific:divider-gray-light"
  >
    <div
      class="flex items-center py-1 transition pl-3 pr-2"
    >
      <div
        class="fill-current w-[1em] h-[1em] inline-block text-lg mr-1"
        data-icon="date_range"
      />
      <input
        aria-label="Datum"
        class=" w-[75px] rounded-sm transition-[outline-color,background-color] bg-transparent font-bold text-center outline-transparent focus-visible:bg-silver focus-visible:outline focus-visible:outline-1 focus-visible:outline-gray-medium focus-visible:outline-offset-0 "
        placeholder="TT.MM.JJJJ"
        value=""
      />
      <button
        aria-label="Kalendar ein-/ausschalten"
        class="transition text-gray-medium hover:text-gray-dark"
      >
        <div
          class="fill-current w-[1em] h-[1em] inline-block text-2xl"
          data-icon="arrow_drop_down"
        />
      </button>
    </div>
  </div>
</div>
`;
